root/types/tablecomponents.js

```diff
@@ -0,0 +1,46 @@
+/*
+ * @flow strict
+ * Copyright (C) 2017 MetaBrainz Foundation
+ *
+ * This file is part of MusicBrainz, the open internet music database,
+ * and is licensed under the GPL version 2, or (at your option) any
+ * later version: http://www.gnu.org/licenses/gpl-2.0.txt
+ */
+
+/*
+ * Types are in alphabetical order.
+ *
+ * The definitions in this file are intended to model the output of the
+ * TO_JSON methods under lib/MusicBrainz/Server/Entity/, those are precisely
+ * how data is serialized for us.
+ */
+
+declare type AppearancesT<T> = {
+  +hits: number,
+  +results: $ReadOnlyArray<T>,
+};
+
+/*
+ * See http://search.cpan.org/~lbrocard/Data-Page-2.02/lib/Data/Page.pm
+ * Serialized in MusicBrainz::Server::TO_JSON.
+ */
+declare type PagerT = {
+  +current_page: number,
+  +first_page: 1,
+  +last_page: number,
+  +next_page: number | null,
+  +previous_page: number | null,
+  +total_entries: number,
+};
+
+declare type SearchResultT<T> = {
```

Comment:
should probably go in root/search/types.js and just imported in the few places it's used. though that just leaves PagerT in this file which is a bit weird